### PR TITLE
shell: xtimerfy sc_icmpv6

### PR DESCRIPTION
**Pull request by authmillenon**
Replaces vtimer by xtimer in the ping6 shell command

### CODE REVIEW

**Comment by authmillenon** [comment]
I looked the values up twice, but please make sure, that they are correct.

**Comment by authmillenon** [comment]
In the meantime: just add `vtimer` to your application's modules.

**Comment by cgundogan** [comment]
On native the numbers look correct. On the testbed - I cannot test it since `PRIu64` doesn't get substituted with the correct value. ACK and GO when travis agrees.

**Comment by OlegHahm** [comment]
Eh, wouldn't it be better to stay with 32 bit values for printing?

**Comment by OlegHahm** [comment]
Please don't merge if this screws up ping6 statistic printing on basically all Cortex-M platforms (because of missing support of printing 64 bit integers in nano-specs).

**Comment by cgundogan** [comment]
>Eh, wouldn't it be better to stay with 32 bit values for printing?

I thought so, too. But this would limit ping statistics to somewhere around ~70 minutes. Longterm pings would wrap then, wouldn't they?

**Comment by OlegHahm** [comment]
Of course, you have to break it down to ms and us before then.

**Comment by cgundogan** [comment]
@authmillenon can you replace all `uint64_t`'s with `uint32_t` and use 2 `uint32_t`'s for the one showing how much time it took overall?

**Comment by authmillenon** [comment]
Done.

File: sys/shell/commands/sc_icmpv6_echo.c [hunk 1]
```diff
@@ -102,13 +102,11 @@ int _handle_reply(gnrc_pktsnip_t *pkt, uint64_t time)
             min_seq_expected++;
         }
 
-        timex_t rt = timex_from_uint64(time);
         printf("%u bytes from %s: id=%" PRIu16 " seq=%" PRIu16 " hop limit=%" PRIu8
-               " time = %" PRIu32 ".%03" PRIu32 " ms\n", (unsigned) icmpv6->size,
+               " time = %" PRIu64 ".%03" PRIu64 " ms\n", (unsigned) icmpv6->size,
```
Comment (cgundogan) [inline sys/shell/commands/sc_icmpv6_echo.c]: here are still some `PRIu64`s

Comment (authmillenon) [inline sys/shell/commands/sc_icmpv6_echo.c]: Fixed

File: sys/shell/commands/sc_icmpv6_echo.c [hunk 2]
```diff
@@ -161,16 +145,16 @@ int _icmpv6_ping(int argc, char **argv)
...
     char *addr_str;
     ipv6_addr_t addr;
     msg_t msg;
     gnrc_netreg_entry_t *ipv6_entry, my_entry = { NULL, ICMPV6_ECHO_REP,
                                                   thread_getpid()
                                                 };
-    timex_t min_rtt = { UINT32_MAX, UINT32_MAX }, max_rtt = { 0, 0 };
-    timex_t sum_rtt = { 0, 0 };
-    timex_t ping_start;
+    uint64_t min_rtt = UINT64_MAX, max_rtt = 0;
```
Comment (cgundogan) [inline sys/shell/commands/sc_icmpv6_echo.c]: `min_rtt` and `max_rtt` can be both `uint32_t`. No way that a round-trip time can take `>UINT32_MAX` microseconds

File: sys/shell/commands/sc_icmpv6_echo.c [hunk 3]
```diff
@@ -119,38 +117,24 @@ int _handle_reply(gnrc_pktsnip_t *pkt, uint64_t time)
...
     if (success > 0) {
-        timex_normalize(&sum_rtt);
+        uint32_t avg_rtt = (uint32_t)sum_rtt / count;  /* get average */
         printf("%d packets transmitted, %d received, %d%% packet loss, time %"
-                PRIu32 ".06%" PRIu32 " s\n", count, success,
-                (100 - ((success * 100) / count)), stop.seconds, stop.microseconds);
-        timex_t avg_rtt = timex_from_uint64(timex_uint64(sum_rtt) / count);  /* get average */
+               PRIu32 ".06%" PRIu32 " s\n", count, success,
+               (100 - ((success * 100) / count)),
+               (uint32_t)total_time / SEC_IN_USEC, (uint32_t)total_time % SEC_IN_USEC);
```
Comment (cgundogan) [inline sys/shell/commands/sc_icmpv6_echo.c]: casting the `total_time` here to `uint32_t` may not show the proper `total_time` if a long-term ping exceeds > ~70 min

Comment (cgundogan) [inline sys/shell/commands/sc_icmpv6_echo.c]: ignore my previous comment about the 70 min mark, but there is still a possibility that `total_time / SEC_IN_USEC` can be greater than `UINT32_MAX`

Comment (cgundogan) [inline sys/shell/commands/sc_icmpv6_echo.c]: well .. if you run the ping for several thousand years at least .. nevermind.

File: sys/shell/commands/sc_icmpv6_echo.c (hunk 2 above)
Comment (authmillenon) [inline sys/shell/commands/sc_icmpv6_echo.c]: Done.

File: sys/shell/commands/sc_icmpv6_echo.c [hunk 4]
```diff
@@ -161,16 +145,16 @@ int _icmpv6_ping(int argc, char **argv)
...
     char *addr_str;
     ipv6_addr_t addr;
     msg_t msg;
     gnrc_netreg_entry_t *ipv6_entry, my_entry = { NULL, ICMPV6_ECHO_REP,
                                                   thread_getpid()
                                                 };
-    timex_t min_rtt = { UINT32_MAX, UINT32_MAX }, max_rtt = { 0, 0 };
-    timex_t sum_rtt = { 0, 0 };
-    timex_t ping_start;
+    uint32_t min_rtt = UINT64_MAX, max_rtt = 0;
```
Comment (cgundogan) [inline sys/shell/commands/sc_icmpv6_echo.c]: s/UINT64_MAX/UINT32_MAX/

**Comment by cgundogan** [comment]
building for `iotlab-m3`:
```
RIOT/examples/gnrc_networking/bin/iotlab-m3/shell_commands.a(sc_icmpv6_echo.o): In function `_icmpv6_ping':
RIOT/sys/shell/commands/sc_icmpv6_echo.c:239: undefined reference to `xtimer_msg_receive_timeout'
collect2: error: ld returned 1 exit status
RIOT/examples/gnrc_networking/../../Makefile.include:213: recipe for target 'all' failed
make: *** [all] Error 1
```

**Comment by authmillenon** [comment]
Unrelated to this PR Oo. It does not exist in `xtimer`.

**Comment by cgundogan** [comment]
you could use the `64` variant for now

**Comment by authmillenon** [comment]
Done

**Comment by cgundogan** [comment]
Compiles now. But:
Every second line of ping statistics has a wrong time value:
```
m3-355;ping6 10 abcd::1 0 0 1
1446565175.363104;m3-355;> ping6 10 abcd::1 0 0 1
1446565175.373257;m3-355;8 bytes from abcd::1: id=83 seq=1 hop limit=64 time = 8.350 ms
1446565175.373481;m3-355;--- abcd::1 ping statistics ---
1446565175.375128;m3-355;1 packets transmitted, 1 received, 0% packet loss, time 0.069767 s
1446565175.376226;m3-355;rtt min/avg/max = 8.350/8.350/8.350 ms
1446565175.385082;m3-355;8 bytes from abcd::1: id=83 seq=2 hop limit=64 time = 7.719 ms
1446565175.386125;m3-355;--- abcd::1 ping statistics ---
1446565175.387105;m3-355;2 packets transmitted, 2 received, 0% packet loss, time 38.0695416 s
1446565175.388078;m3-355;rtt min/avg/max = 7.719/8.034/8.350 ms
1446565175.399314;m3-355;8 bytes from abcd::1: id=83 seq=3 hop limit=64 time = 9.299 ms
1446565175.399530;m3-355;--- abcd::1 ping statistics ---
1446565175.401278;m3-355;3 packets transmitted, 3 received, 0% packet loss, time 0.0623456 s
1446565175.402246;m3-355;rtt min/avg/max = 7.719/8.456/9.299 ms
1446565175.413290;m3-355;8 bytes from abcd::1: id=83 seq=4 hop limit=64 time = 9.297 ms
1446565175.413533;m3-355;--- abcd::1 ping statistics ---
1446565175.415161;m3-355;4 packets transmitted, 4 received, 0% packet loss, time 38.06109085 s
1446565175.416242;m3-355;rtt min/avg/max = 7.719/8.666/9.299 ms
1446565175.427162;m3-355;8 bytes from abcd::1: id=83 seq=5 hop limit=64 time = 9.928 ms
1446565175.428261;m3-355;--- abcd::1 ping statistics ---
1446565175.429267;m3-355;5 packets transmitted, 5 received, 0% packet loss, time 0.0637794 s
1446565175.430162;m3-355;rtt min/avg/max = 7.719/8.918/9.928 ms
1446565175.440527;m3-355;8 bytes from abcd::1: id=83 seq=6 hop limit=64 time = 8.035 ms
1446565175.440739;m3-355;--- abcd::1 ping statistics ---
1446565175.442217;m3-355;6 packets transmitted, 6 received, 0% packet loss, time 38.06121491 s
1446565175.443181;m3-355;rtt min/avg/max = 7.719/8.771/9.928 ms
1446565175.454139;m3-355;8 bytes from abcd::1: id=83 seq=7 hop limit=64 time = 9.613 ms
1446565175.455136;m3-355;--- abcd::1 ping statistics ---
1446565175.456107;m3-355;7 packets transmitted, 7 received, 0% packet loss, time 0.0651817 s
1446565175.457135;m3-355;rtt min/avg/max = 7.719/8.891/9.928 ms
1446565175.466014;m3-355;8 bytes from abcd::1: id=83 seq=8 hop limit=64 time = 7.721 ms
1446565175.467094;m3-355;--- abcd::1 ping statistics ---
1446565175.468193;m3-355;8 packets transmitted, 8 received, 0% packet loss, time 38.06133583 s
1446565175.469051;m3-355;rtt min/avg/max = 7.719/8.745/9.928 ms
1446565175.480058;m3-355;8 bytes from abcd::1: id=83 seq=9 hop limit=64 time = 9.299 ms
1446565175.480948;m3-355;--- abcd::1 ping statistics ---
1446565175.482038;m3-355;9 packets transmitted, 9 received, 0% packet loss, time 0.0665527 s
1446565175.483034;m3-355;rtt min/avg/max = 7.719/8.806/9.928 ms
1446565175.494946;m3-355;8 bytes from abcd::1: id=83 seq=10 hop limit=64 time = 9.928 ms
1446565175.495149;m3-355;--- abcd::1 ping statistics ---
1446565175.496977;m3-355;10 packets transmitted, 10 received, 0% packet loss, time 38.06147896 s
1446565175.498036;m3-355;rtt min/avg/max = 7.719/8.918/9.928 ms
```

**Comment by cgundogan** [comment]
the "time jump" also appears with greater values than 1 for printing the ping statistics. Every second call to the print statistics function results in a wrong time value

**Comment by authmillenon** [comment]
Last time I tested it (first commit of this PR it worked)… will look into it tomorrow.

**Comment by authmillenon** [comment]
Fixed

**Comment by cgundogan** [comment]
works on testbed and on native. ACK

**Comment by authmillenon** [comment]
Squashed

**Comment by cgundogan** [comment]
travis is happy. GO

**Pull request merged**